stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -80,6 +79,29 @@ class TaskFutureWaitAsyncContext : public AsyncContext {
   using AsyncContext::AsyncContext;
 };
 
+/// An asynchronous context used by `TaskGroup.next()` describing a "Future".
+///
+/// This type matches the ABI of a function `<T> () async throws -> T`,
+/// which is the type used by `TaskGroup.next()`.
+class TaskGroupNextWaitAsyncContext : public AsyncContext {
+public:
+  // Error result is always present.
+  SwiftError *errorResult = nullptr;
+
+  // No indirect results.
+
+  TaskFutureWaitResult result;
+
+  // FIXME: Currently, this is always here, but it isn't technically necessary.
+  void* Self;
+
+  // Arguments.
+  AsyncTask *task;
+  TaskGroup *group;
+
+  using AsyncContext::AsyncContext;
+};
```

Comment:
we need this because when we cast the context the C await next impl is getting, we must have the `group` at hand; and the waiting task is `task` so it does not have the group stored anymore, so we carry it here.

This reflects the `swift_task_group_wait_next` functions signature